packages/@aws-cdk/aws-servicecatalog/lib/product.ts

```diff
@@ -0,0 +1,212 @@
+import { ArnFormat, IResource, Resource, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CloudFormationTemplate } from './cloudformation-template';
+import { AcceptLanguage } from './common';
+import { InputValidator } from './private/validation';
+import { CfnCloudFormationProduct } from './servicecatalog.generated';
+
+/**
+ * A Service Catalog product, currently only supports type CloudFormationProduct
+ */
+export interface IProduct extends IResource {
+  /**
+   * The ARN of the product.
+   * @attribute
+   */
+  readonly productArn: string;
+
+  /**
+   * The id of the product
+   * @attribute
+   */
+  readonly productId: string;
+}
+
+abstract class ProductBase extends Resource implements IProduct {
+  public abstract readonly productArn: string;
+  public abstract readonly productId: string;
+}
+
+/**
+ * Properties of product version (also known as a provisioning artifact).
+ */
+export interface CloudFormationProductVersion {
+  /**
+   * The description of the product version
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * Whether the specified product template will be validated by CloudFormation.
+   * If turned off, an invalid template configuration can be stored.
+   * @default true
+   */
+  readonly validateTemplate?: boolean;
+
+  /**
+   * The S3 template that points to the provisioning version template
+   */
+  readonly cloudFormationTemplate: CloudFormationTemplate;
+
+  /**
+   * The name of the product version.
+   * @default - No product version name provided
+   */
+  readonly productVersionName?: string;
+}
+
+/**
+ * Properties for a Cloudformation Product
+ */
+export interface CloudFormationProductProps {
+  /**
+   * The owner of the product.
+   */
+  readonly owner: string;
+
+  /**
+   * The name of the product.
+   */
+  readonly productName: string;
+
+  /**
+   * The configuration of the product version.
+   */
+  readonly cloudFormationProductVersions: CloudFormationProductVersion[];
```

Comment:
I actually liked the name `productVersions` more 😕.

Even better, I like the name `versions` even more than `productVersions` here - I don't think that's ambiguous, unlike a property called `name`.

Can you change it? I'll leave the decision whether to go with `versions`, or `productVersions`, to your judgement.

Reply:
lets just go with productVersions then since that's what we had originally